### PR TITLE
Add spec text

### PR DESCRIPTION
**Pull request by jridgewell**

### CODE REVIEW

**Comment by jridgewell** [comment]
Merging so that the spec text can be viewed. Please still add reviews, and I'll address in follow ups.

File: spec.emu [hunk 1]
```diff
@@ -4,20 +4,308 @@
...
+      <p>When called with a template and a variable number of substitutions, a string is returned which has common leading indentation removed from all lines with non-whitespace.</p>
+      <p>When called with a function, a closure is returned which wraps the function. When the closure is called with a template and variable number of substitutions, the common leading indentation of the temlpate is removed and the result of calling the function with this new dedented template and substitutions is returned.</p>
+      <p>Common leading indentation is defined as the longest sequence of whitespace characters that match exactly on every line that contains non-whitespace. Empty lines and lines which contain only whitespace do not affect common leading indentation.</p>
+      <p>When removing the common leading indentation, lines which contain only whitespace will have all whitespace removed.</p>
+    </emu-note>
+    <emu-alg>
+      1. If Type(_templateOrFn_) is Object and IsCallable(_templateOrFn_) is *true*, then
```
Comment (ljharb) [inline spec.emu:37]: Can the type not be object, while IsCallable is true?

Comment (jridgewell) [reply spec.emu:37]: Actually, `IsCallable` checks type is `Object`. I'll remove.

Comment (jridgewell) [reply spec.emu:37]: af8443f

**Comment by ljharb** [comment]
The spec also always throws in the callable branch - 1.b. makes an Abstract Closure and does nothing with it.

**Comment by ljharb** [comment]
Also, LeadingQhitespaceString references an S that doesn’t exist - and I’m not sure the prose is clear enough to explain what the AO should be doing. It’d probably be better to do something more explicit.

**Comment by jridgewell** [comment]
> The spec also always throws in the callable branch - 1.b. makes an Abstract Closure and does nothing with it.

Fixed locally. I have a large refactor coming to split the huge `DedentStringsArray` into smaller AOs.

> Also, LeadingWhitespaceSubstring references an S that doesn’t exist

Fixed.



> and I’m not sure the prose is clear enough to explain what the AO should be doing. It’d probably be better to do something more explicit.

I took this wording from [TrimString](https://tc39.es/ecma262/multipage/text-processing.html#sec-trimstring). I'm open to anything that lets me say "if this is Unicode white space".

> IsAllWhitespace’s name says to me that it’s true when the string contains nothing but whitespace

That's the intention.

**Comment by jridgewell** [comment]
Pushed AOs and fixes in [f9d3621](https://github.com/tc39/proposal-string-dedent/commit/f9d36218db56838da4889c1e6ba88d7a41385a6f).

**Comment by ljharb** [comment]
If that's the intention, can it just _call_ TrimString and assert that it receives an empty string back?

**Pull request merged**